Subcell GRMHD: add VariablesNeededFixing tag and TCI options

nilsdeppe opened the pull request:
## Proposed changes

- The `VariablesNeededFixing` is used when on the subcells to tell the TCI if the solution was bad and we shouldn't go back to DG. 
- And `TciOptions` struct that houses the system-specific TCI options. At some point it might be worth revisiting the input file layout for all the subcell code, but I think that's best done after we can do evolutions and know what features we need.

### Upgrade instructions

<!--
If this PR makes changes that other people should be aware of when upgrading
their code, describe what they should do between the two UPGRADE INSTRUCTIONS
lines below.
-->
<!-- UPGRADE INSTRUCTIONS -->

<!-- UPGRADE INSTRUCTIONS -->

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `major new feature` if appropriate.

### Further comments

<!--
If this is a relatively large or complex change, kick off the discussion by
explaining why you chose the solution you did and what alternatives you
considered, etc...
-->

nilsdeppe committed:
Add GRMHD VariablesNeededFixing tag

nilsdeppe committed:
Add ValenciaDivClean TCI options

kidder commented on src/Evolution/Systems/GrMhd/ValenciaDivClean/Subcell/TciOptions.hpp:
```diff
@@ -0,0 +1,89 @@
+        "DG element is below this value we skip primitive recovery and treat "
+        "the cell as atmosphere."};
+  };
+  /// \brief Safety factor \f$\epsilon_B\f$.
```
Please include a bit more information; I have no idea reading this if it should be big or small (it's fine to point to some other more detailed documentation)

nilsdeppe replied:
I can't link to the `TciOnDgGrid` yet because it's not merged. I'll add a sentence that this is where the documentation will be

fmahebert commented:
I have no additional requests

nilsdeppe commented:
I've rebased and squashed in the additional sentence pointing to the (not yet merged) documentation of the TciOnDgGrid mutator. Thanks for the reviews :)
